tests/remote/fetch.c

```diff
@@ -0,0 +1,184 @@
+#include "../clar_libgit2.h"
+
+#include "remote.h"
+#include "repository.h"
+
+static git_repository *repo1;
+static git_repository *repo2;
+static char* repo1_path;
+static char* repo2_path;
+
+static const char *REPO1_REFNAME = "refs/heads/main";
+static const char *REPO2_REFNAME = "refs/remotes/repo1/main";
+static char *FORCE_FETCHSPEC = "+refs/heads/main:refs/remotes/repo1/main";
+static char *NON_FORCE_FETCHSPEC = "refs/heads/main:refs/remotes/repo1/main";
+
+char* strip_trailing_slash(char *path)  {
+	if (path[strlen(path) - 1] == '/') {
+		char* result = (char *) calloc(strlen(path) - 1, sizeof(char));
+		memcpy(result, path, strlen(path) - 1);
+		return result;
+	} else {
+		char* result = (char *) calloc(strlen(path), sizeof(char));
+		strncpy(result, path, strlen(path));
+		return result;
+	}
+}
+
+
+void test_remote_fetch__initialize(void) {
+	git_config *c;
+	git_buf repo1_path_buf = GIT_BUF_INIT;
+	git_buf repo2_path_buf = GIT_BUF_INIT;
+	const char *sandbox = clar_sandbox_path();
+
+	cl_git_pass(git_buf_join(&repo1_path_buf, '/', sandbox, "fetchtest_repo1"));
```

Comment:
`git_buf_joinpath` would be nicer here; it's what I was expecting and made me think you were trying to prepend slash :/

Reply:
Done